addons/modules/functions/fnc_moduleSpawnReinforcements.sqf

```diff
@@ -0,0 +1,136 @@
+createVehicleCrew _vehicle;
+
+crew _vehicle joinSilent _vehicleGroup;
+_vehicleGroup addVehicle _vehicle;
+
+_vehicleGroup selectLeader commander _vehicle;
+_vehicleGroup allowFleeing 0;
+
+// Force aircraft to fly at the correct height and ignore surroundings
+if (_isAir) then {
+    _vehicleGroup setBehaviour "CARELESS";
+    _vehicle flyInHeight AIRCRAFT_HEIGHT;
+};
+
+// Increase the skill of the driver/pilot for better driving/flying
+driver _vehicle setSkill 1;
+
+// Spawn the infantry units and move them into the vehicle's cargo positions
+private _infantryGroup = createGroup [_side, true];
+
+{
+    private _unit = _infantryGroup createUnit [_x, [0, 0, 0], [], 0, "CAN_COLLIDE"];
+    _unit moveInCargo _vehicle;
+} forEach _infantryTypes;
+
+// Use normal transport unload if fastroping is not available
+if (_insertionMethod == 2 && {!isClass (configFile >> "CfgPatches" >> "ace_fastroping") || {getNumber (_vehicleConfig >> "ace_fastroping_enabled") == 0}}) then {
+    _insertionMethod = 0;
+};
+
+// Add waypoint to make vehicle drop units off at the LZ
+// Handle insertion method parameter for air vehicles
+private _waypoint = _vehicleGroup addWaypoint [_positionLZ, WAYPOINT_RADIUS];
+
+if (_isAir && {_insertionMethod > 0}) then {
+    private _script = [
+        QPATHTOEF(ai,functions\fnc_waypointParadrop.sqf),
+        QPATHTOEF(ai,functions\fnc_waypointFastrope.sqf)
+    ] select (_insertionMethod == 2);
+
+    _waypoint setWaypointType "SCRIPTED";
+    _waypoint setWaypointScript _script;
+} else {
+    _waypoint setWaypointType "TR UNLOAD";
+};
+
+// Add waypoint to make infantry units move to RP after they arrive at LZ
+if (_positionRP isEqualType []) then {
+    _infantryGroup addWaypoint [_positionRP, WAYPOINT_RADIUS];
+};
+
+// Adjust behaviour of infantry units if not default
+if (_infantryBehaviour > 0) then {
+    private _behaviour = ["SAFE", "AWARE", "COMBAT"] select (_infantryBehaviour - 1);
+    private _speedMode = ["LIMITED", "NORMAL"] select (_infantryBehaviour > 1);
+
+    _infantryGroup setBehaviour _behaviour;
+    _infantryGroup setSpeedMode _speedMode;
+};
+
+// Add waypoint to make vehicle return to spawn position and despawn if needed
+if (_despawnVehicle) then {
+    private _waypoint = _vehicleGroup addWaypoint [_spawnPosition, WAYPOINT_RADIUS];
+    _waypoint setWaypointStatements ["true", "deleteVehicle vehicle this; {deleteVehicle _x} forEach thisList"];
+    _waypoint setWaypointTimeout [5, 7.5, 10];
+} else {
+    // Otherwise make aircraft stay around the LZ and provide air support
+    if (_isAir) then {
+        private _waypoint = _vehicleGroup addWaypoint [_positionLZ, WAYPOINT_RADIUS];
+        _waypoint setWaypointType "SAD";
+        _waypoint setWaypointBehaviour "AWARE";
+        _waypoint setWaypointCombatMode "RED";
+    };
+};
+
+// Add the vehicle and crew + passengers to curators
+[QEGVAR(common,addObjects), [[_vehicle]]] call CBA_fnc_localEvent;
```

Comment:
return vehicle, crew and cargo would be nice if someone like me want to use it, but it is not public
`[_vehicle, _vehicleGroup, _infantryGroup]`

Reply:
The function is not supposed to be exposed to the public API, it only is called from the confirm button of the module. So no return value is needed.